app/src/docs/hooks/useLunr.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import { useState, useEffect, useMemo } from 'react'
+import lunr from 'lunr'
+
+const InvalidIndexError = Error('Lunr index could not be parsed. Check that your index exists and is valid.')
+const InvalidStoreError = Error('Lunr store could not be parsed. Check that your store exists and is valid.')
```

Comment:
Note that throwing these errors won't include appropriate stack traces, the stack trace is generated when the `new Error()` is created and these are created during initialisation, rather than when the error occurs

Reply:
Could do something like 

```js
const InvalidIndexError = () => new Error(…)
// then
throw InvalidIndexError() // generates new error object with appropriate stack trace
```